src/main/java/org/ngafid/GenerateBulkCSVS.java

```diff
@@ -383,7 +383,7 @@ public static void main(String[] args) {
 			System.exit(1);
 		}
 
-		if (!dir.substring(dir.length() - 1, dir.length()).equals("/")) {
+		if (dir.charAt(dir.length() - 1) != '/') {
```

Comment:
revert this too - instead it would be better to throw an exception if the directory length is 0